backend/ActivitiesAnalysis/fetchUserData.js

```diff
@@ -0,0 +1,83 @@
+const express = require('express');
+const router = express.Router();
+const { PrismaClient } = require('@prisma/client');
+const prisma = new PrismaClient();
+const authenticateToken = require('../MiddleWare/authenticateToken');
+const { categorizeMood } = require('../DataAnalysis/analysis');
+
+// fetch the user's activities for the previous week
+const fetchUserActivity = async (userId) => {
+    try {
+        const activities = await prisma.userEntry.findMany({
+            where: {
+                userId: userId,
+                date: {
+                    gte: new Date(Date.now() - 7 * 24 * 60 * 60 * 1000),
```

Comment:
nit: If you are using this 7 * 24 * 60 * 60 * 1000 number multiple times, may be good to just turn it into a const variable 

like const DATE_RANGE = 7 * 24 * 60 * 60 * 1000 or something, and share it throughout this file.

Reply:
That sound's like a good idea. I'll implement that